fundingmanager_test.go

```diff
@@ -0,0 +1,596 @@
+	var err error
+	miningNode, err = rpctest.New(netParams, nil, nil)
+	if err != nil {
+		t.Fatalf("unable to create mining node: %v", err)
+	}
+	if err = miningNode.SetUp(true, 25); err != nil {
+		t.Fatalf("unable to set up mining node: %v", err)
+	}
+
+	// Next mine enough blocks in order for segwit and the CSV package
+	// soft-fork to activate on SimNet.
+	numBlocks := netParams.MinerConfirmationWindow * 2
+	if _, err = miningNode.Node.Generate(numBlocks); err != nil {
+		t.Fatalf("unable to generate blocks: %v", err)
+	}
+
+	rpcConfig := miningNode.RPCConfig()
+
+	chainNotifier, err := btcdnotify.New(&rpcConfig)
+	if err != nil {
+		t.Fatalf("unable to create notifier: %v", err)
+	}
+	if err = chainNotifier.Start(); err != nil {
+		t.Fatalf("unable to start notifier: %v", err)
+	}
+	aliceTestDir, err = ioutil.TempDir("", "alicelnwallet")
+	if err != nil {
+		t.Fatalf("unable to create temp directory: %v", err)
+	}
+
+	aliceMsgChan = make(chan lnwire.Message)
+	aliceAnnounceChan = make(chan lnwire.Message)
+	aliceFundingMgr, err = createTestFundingManager(t, alicePubKey, aliceTestDir, alicePrivKeyBytes[:], netParams, rpcConfig, miningNode, chainNotifier, aliceMsgChan, aliceAnnounceChan)
+	if err != nil {
+		t.Fatalf("failed creating fundingManager: %v", err)
+	}
+	if err = aliceFundingMgr.Start(); err != nil {
+		t.Fatalf("failed starting fundingManager: %v", err)
+	}
+
+	bobTestDir, err = ioutil.TempDir("", "boblnwallet")
+	if err != nil {
+		t.Fatalf("unable to create temp directory: %v", err)
+	}
+	bobMsgChan = make(chan lnwire.Message)
+	bobAnnounceChan = make(chan lnwire.Message)
+	bobFundingMgr, err = createTestFundingManager(t, bobPubKey, bobTestDir, bobPrivKeyBytes[:], netParams, rpcConfig, miningNode, chainNotifier, bobMsgChan, bobAnnounceChan)
+	if err != nil {
+		t.Fatalf("failed creating fundingManager: %v", err)
+	}
+	if err = bobFundingMgr.Start(); err != nil {
+		t.Fatalf("failed starting fundingManager: %v", err)
+	}
+}
+
+func tearDownFundingManagers() {
+	os.RemoveAll(bobTestDir)
+	os.RemoveAll(aliceTestDir)
+	miningNode.TearDown()
+}
+
+func TestFundingManager_normalWorkflow(t *testing.T) {
+	setupFundingManagers(t)
+	defer tearDownFundingManagers()
+
+	// Create a funding request and start the workflow
+	errChan := make(chan error, 1)
+	updateChan := make(chan *lnrpc.OpenStatusUpdate, 1)
+	initReq := &openChanReq{
+		targetPeerID:    int32(1),
+		targetPubkey:    bobPubKey,
+		localFundingAmt: 500000,
+		pushAmt:         0,
+		numConfs:        1,
+		updates:         updateChan,
+		err:             errChan,
+	}
+
+	aliceFundingMgr.initFundingWorkflow(bobAddr, initReq)
+
+	// Alice should have sent the init message to Bob
+	fundingReq := <-aliceMsgChan
+	if fundingReq.MsgType() != lnwire.MsgSingleFundingRequest {
+		t.Fatalf("expected MsgSingleFundingRequest to be sent from alice")
+	}
+
+	// Let Bob handle the init message
+	bobFundingMgr.processFundingRequest(fundingReq.(*lnwire.SingleFundingRequest), aliceAddr)
```

Comment:
As is now, if the `fundingReq` (and other instances below), _don't_ aren't actually the specified type, then the test will panic. Instead, the interface should be re-assigned to a new variable, with a two-value assignment so the type assertion can be verified: 
```
fundingReqMsg, ok := fundingReq.(*lnwire.SingleFundingRequest)
if !ok {
    t.Fatalf("funding request message not sent, instead %T sent: fundingReq")
}
```